### DISCUSSION TITLE
Comma as thousand seperator for decimal number

### DISCUSSION THREAD

**Discussion by mikehZA** [micropython/micropython#15928] [Using MicroPython]
Hi I am using Pi Pico MicroPython v1.19.1 I cannot find a way to get decimal numbers and thousand seperator looks like they are mutually exclusive.
```
number = 456324
print('{:,}'.format(number))  # Works

print('{:,.2f}'.format(number)) # Not working

number = 456324.56
print('{:,}'.format(number))  # Not working

print('{:,.2f}'.format(number))   # Not working
```
any suggestions

**Comment by mattytrentini** [comment]
It's a known issue, please see #11284.

**Reply by mikehZA** [reply]
Ok thanks, so looking at the dates I guess it is not regarded as something important

**Reply by mattytrentini** [reply]
I think it's fair to say that it's considered a low priority for the core devs (it doesn't cause a crash and there are - admittedly awkward - workarounds). That said, it'd be _very_ welcome patch!

**Reply by mattytrentini** [reply]
The file that would need to be modified appears to be [formatfloat.c](https://github.com/micropython/micropython/blob/master/py/formatfloat.c) and tests could be added to [tests/float/float_format.py](https://github.com/micropython/micropython/blob/master/tests/float/float_format.py). Tests could include the following:

```diff
diff --git a/tests/float/float_format.py b/tests/float/float_format.py
index 98ed0eb096..7776b6ea4f 100644
--- a/tests/float/float_format.py
+++ b/tests/float/float_format.py
@@ -25,3 +25,7 @@ for r in range(38):
     # But formatting as 0.999...e-r is NOT ok.
     if s[0] == "0":
         print("FAIL:", s)
+
+# Test the thousands separator
+for val in (0, 1, 99, 1000, 9999, 456324, 1_000_000):
+    print("{:,f}".format(float(val)))
```
